DOC-2639-4.2.0 bug fixes (#786)
tigergraph/server-docs: modules/release-notes/pages/index.adoc

== Fixed issues

=== Fixed and Improved in 4.2.0

==== Functionality

* Fixed issue where the loading job status was incorrectly cleaned up in advance due to an inaccurate loading job counter (TP-4741).
* Fixed issue where the loader would fail silently, and now it reports an error directly if the loader fails to start (TP-6420).
* Fixed issue where loading an empty file triggered a failure loop (TP-6530).

> * Added the 'graph' field to CDC messages generated by the TigerGraph CDC service (CORE-4147).

Reviewer: * Added the 'graph' field to CDC messages generated by the TigerGraph CDC service (CORE-4147).
This one should be mentioned in Compatibility Issues also.

* Fixed issue with audit error codes being logged when a CDC message fails to deliver to external Kafka (CORE-4249).
* Fixed critical disk issue caused by the rebuilder getting stuck in a partitioned cluster after dropping vertex or edge attributes (CORE-4303).
* Fixed issue where CDC EDGE messages with "UNKNOWN" ID were not removed when a query deleted a vertex and edge simultaneously (CORE-4457).
* Fixed issue where the primary ID was missing when inserting a vertex implicitly from loading an edge in a single-partition cluster and the target vertex used "primary_id_as_attribute=true" (GLE-7562).

> * Fixed issue with running the TigerGraph command `gcollect` on Kubernetes (TP-6353).

Reviewer: * Fixed issue with running the TigerGraph command `gcollect` on Kubernetes (TP-6353).

This one doesn't tell us how the user experienced the problem (the symptom).

> * Added support for debug mode when a pod restart fails due to a PostStartHookError (TP-7228).

Reviewer: * Added support for debug mode when a pod restart fails due to a PostStartHookError (TP-7228).
What pod? Kubernetes or Savanna?

"pod" is a Kubernetes concept. Is this for the on-prem product or for savanna (which used Kubernetes under the hood)?

> * Added namespace as a suffix of the HostName in HostList on Kubernetes (TP-6214).

Reviewer: The ticket does not explain why this was done. Is this fixing a problem or an "improvement" to provide better information? Is it a compatibility issue (4.1 vs 4.2)?

Apparently this was changed in 4.1.1 first, but we didn't mention it them.

> * Fixed issue with query compilation errors when iterating over a foreach statement to retrieve an accum type attribute with the `-single` or `distributed` keyword, and ensured that operations between two different `GroupByAccums` now hold the same data types and field names (GLE-8507).

Reviewer: ```suggestion
* Fixed query compilation errors when iterating over `FOREACH` to retrieve an accum type attribute with the `-single` or `distributed` keyword; ensured operations between two different `GroupByAccums` hold the same data types and field names (GLE-8507).
```

* Fixed issue with double quotes in literal strings during the creation of a loading job (GLE-8630).

> * Fixed issue with fetching dynamic file output policy during query execution (GLE-4847).

Reviewer: Neither of these explain the nature of the error.

Hypothetically, if the "issue with fetching dynamic file output policy during query execution" was the inability to perform the fetch, then a better message would be
"Fixed the inability to fetch dynamic file output policy during query execution (GLE-4847)."

In the phrase "Fixed issue", "issue" is an empty word. It tells us nothing.

* Fixed issue where sensitive information was not removed from the browser localStorage, ensuring better security (APPS-1066)

> * Fixed issue with query installation failure for single-node queries that initialize vertex set variables in conditional branches, such as `if-else` or `case-when` statements (GLE-7369).

Reviewer: ```suggestion
* Fixed a query installation failure for single-node queries that initialize vertex set variables in conditional branches, such as `if-else` or `case-when` statements (GLE-7369).
```
